src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,210 @@
+                            list.size()));
+                    ui.printline();
+                }
+            } else if (text.startsWith("deadline")) {
+                String[] splitText = text.split("/", 2);
+
+                String description = splitText[0].substring(8);
+                if (description.isEmpty()) {
+                    throw new DukeException("I apologise, sir. " +
+                            "But the description and deadline cannot be empty");
+                } else {
+
+                    String deadlineText = splitText[1].substring(3);
+                    try {
+                        LocalDateTime deadline = LocalDateTime.parse(deadlineText);
+                        Deadline dl = new Deadline(description.trim(), deadline);
+                        list.add(dl);
+
+                        ui.printline();
+                        ui.sendMessage("Noted Sir. I've added this task to your list: ");
+                        ui.sendMessage(String.format("\t [%s] [%s] %s", dl.tag, dl.getStatusIcon(),
+                                dl.toString()));
+                        ui.sendMessage(String.format("As of now, you have %d tasks on the agenda.",
+                                list.size()));
+                        ui.printline();
+                    } catch (DateTimeParseException e) {
+                        throw new DukeException("Invalid Date Format: should be YYYY-MM-DDTHH:MM:SS. " +
+                                "Example: 2023-12-12T06:30:00");
+                    }
+                    try {
+                        storage.appendToFile(text + "\n");
+                    } catch (IOException e) {
+                        System.out.println("Something went wrong: " + e.getMessage());
+                    }
+                }
+            } else if (text.startsWith("event")) {
+                String[] splitText = text.split("/");
+                String description = splitText[0].substring(5);
+                if (description.isEmpty()) {
+                    throw new DukeException("I apologise, sir. " +
+                            "But the description, start and end cannot be empty");
+
+                } else {
+                    String startText = splitText[1].trim().substring(5);
+                    String endText = splitText[2].trim().substring(3);
+
+                    try {
+                        LocalDateTime start = LocalDateTime.parse(startText);
+                        LocalDateTime end = LocalDateTime.parse(endText);
+                        Event event = new Event(description.trim(), start, end);
+                        list.add(event);
+
+                        ui.printline();
+                        ui.sendMessage("Noted Sir. I've added this task to your list: ");
+                        ui.sendMessage(String.format("\t [%s] [%s] %s", event.tag,
+                                event.getStatusIcon(),
+                                event.toString()));
```

Comment:
Can `getStatusIcon()` be included as a part of the `toString()` method? ie the `toString()` method for `ToDo` can return [X] description. Instead of just returning the description.